features/build/buildlogic.feature

```diff
@@ -258,65 +258,173 @@ Feature: buildlogic.feature
+    Given I run the steps 6 times:
+    """
+    When I run the :start_build client command with:
+      | buildconfig | myphp |
+    Then the step should succeed
+    """
+    Given the "myphp-7" build completed
+    Then I wait up to 120 seconds for the steps to pass:
+    """
+    When I save project builds into the :builds_all clipboard
+    And evaluation of `cb.builds_all.select{|b| b.status?(user: user, status: :complete)[:success]}.size` is stored in the :builds_nums clipboard
+    Then the expression should be true> cb.builds_nums >= 5 and cb.builds_nums < 7
+    """
+    Given I get project builds
+    Then the output should not contain:
+      | myphp-1 |
+
+  # @author xiuwang@redhat.com
+  # @case_id OCP-24154
+  # Split 13914 mapping the e2e test scripts
+  Scenario: Should prune canceled builds based on the failedBuildsHistoryLimit setting	
     Given I have a project
     When I run the :new_app client command with:
-      | image_stream | ruby                                          |
-      | code         | https://github.com/openshift/ruby-hello-world |
+      | template | rails-postgresql-example |
     Then the step should succeed
-    When I run the :get client command with:
-      | resource      | buildconfig      |
-      | resource_name | ruby-hello-world |
-      | o             | yaml             |
+    Given I run the :patch client command with:
+      | resource      | bc                                       |
+      | resource_name | rails-postgresql-example                 |
+      | p             | {"spec":{"failedBuildsHistoryLimit": 2}} |
+    Then the step should succeed
+    When I run the :cancel_build client command with:
+      | build_name | rails-postgresql-example-1 |
     Then the step should succeed
-    And the expression should be true> @result[:parsed]["spec"]["failedBuildsHistoryLimit"] == 5
-    And the expression should be true> @result[:parsed]["spec"]["successfulBuildsHistoryLimit"] == 5
-    Given the "ruby-hello-world-1" build completed
-    Given I run the steps 5 times:
+    Given I run the steps 3 times:
     """
     When I run the :start_build client command with:
-      | buildconfig | ruby-hello-world |
+      | buildconfig | rails-postgresql-example |
     Then the step should succeed
     """
-    Given the "ruby-hello-world-6" build completed
+    When I run the :cancel_build client command with:
+      | bc_name | bc/rails-postgresql-example |
+    Then the step should succeed
     Then I wait up to 480 seconds for the steps to pass:
     """
     Given I get project builds
-    Then the output should match 5 times:
-      | Complete |
+    Then the output should match 2 times:
+      | Git.*Cancelled |
     Then the output should not contain:
-      |ruby-hello-world-1|
+      | rails-postgresql-example-1 |
     """
-    Given I run the steps 3 times:
+
+  # @author xiuwang@redhat.com
+  # @case_id OCP-24155
+  # Split 13914 mapping the e2e test scripts
```

Comment:
Do we need this comment and other similar ones? After the split we have separate test cases and nobody would care what the original one was. Or not?

Reply:
otherwise lg, thanks

Reply:
Right, nobody cares the original one. I will remove these comments

Reply:
Only keep comment for the original case OCP-13914.